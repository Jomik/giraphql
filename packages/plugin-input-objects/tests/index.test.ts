import { execute, lexicographicSortSchema, printSchema } from 'graphql';
import { gql } from 'graphql-tag';
import schema from './examples/starwars/schema';

describe('input objects', () => {
  it('generates expected schema', () => {
    expect(printSchema(lexicographicSortSchema(schema))).toMatchSnapshot();
  });

  describe('queries', () => {
    it('uses default input name', async () => {
      const query = gql`
        query {
          human(input: { id: "1002" }) {
            name
          }
        }
      `;

      const result = await execute({ schema, document: query });

      expect(result).toMatchInlineSnapshot(`
Object {
  "data": Object {
    "human": Object {
      "name": "Han Solo",
    },
  },
}
`);
    });

    it('uses custom options name', async () => {
      const query = gql`
        query {
          hero(options: { episode: NEWHOPE }) {
            name
          }
        }
      `;

      const result = await execute({ schema, document: query });

      expect(result).toMatchInlineSnapshot(`
Object {
  "data": Object {
    "hero": Object {
      "name": "R2-D2",
    },
  },
}
`);
    });
  });
});
