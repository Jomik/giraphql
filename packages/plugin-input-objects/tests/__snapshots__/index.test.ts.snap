// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input objects generates expected schema 1`] = `
"interface Character {
  \\"\\"\\"Which movies they appear in.\\"\\"\\"
  appearsIn: [Episode!]!

  \\"\\"\\"The friends of the character, or an empty list if they have none.\\"\\"\\"
  friends: [Character]!

  \\"\\"\\"The id of the character\\"\\"\\"
  id: ID!

  \\"\\"\\"The name of the character\\"\\"\\"
  name: String!
}

\\"\\"\\"A mechanical creature in the Star Wars universe.\\"\\"\\"
type Droid implements Character {
  \\"\\"\\"Which movies they appear in.\\"\\"\\"
  appearsIn: [Episode!]!

  \\"\\"\\"The friends of the character, or an empty list if they have none.\\"\\"\\"
  friends: [Character]!

  \\"\\"\\"The id of the character\\"\\"\\"
  id: ID!

  \\"\\"\\"The name of the character\\"\\"\\"
  name: String!

  \\"\\"\\"The primary function of the droid.\\"\\"\\"
  primaryFunction: String!
}

\\"\\"\\"One of the films in the Star Wars Trilogy\\"\\"\\"
enum Episode {
  \\"\\"\\"Released in 1980.\\"\\"\\"
  EMPIRE

  \\"\\"\\"Released in 1983\\"\\"\\"
  JEDI

  \\"\\"\\"Released in 1977.\\"\\"\\"
  NEWHOPE
}

input HeroInput {
  \\"\\"\\"
  If omitted, returns the hero of the whole saga. If provided, returns the hero of that particular episode.
  \\"\\"\\"
  episode: Episode!
}

\\"\\"\\"A humanoid creature in the Star Wars universe.\\"\\"\\"
type Human implements Character {
  \\"\\"\\"Which movies they appear in.\\"\\"\\"
  appearsIn: [Episode!]!

  \\"\\"\\"The friends of the character, or an empty list if they have none.\\"\\"\\"
  friends: [Character]!

  \\"\\"\\"The home planet of the human, or null if unknown.\\"\\"\\"
  homePlanet: String

  \\"\\"\\"The id of the character\\"\\"\\"
  id: ID!

  \\"\\"\\"The name of the character\\"\\"\\"
  name: String!
}

input HumanInput {
  \\"\\"\\"id of the character\\"\\"\\"
  id: ID!
}

type Query {
  hero(options: HeroInput!): Character!
  human(input: HumanInput!): Human!
  r2d2: Droid!
}
"
`;
